dask/blockwise.py

```diff
@@ -252,6 +252,9 @@ def _out_numblocks(self):
 
         return out_d
 
+    def is_materialized(self):
+        return hasattr(self, "_cached_dict")
```

Comment:
What are these for?  Long-term, I would hope that we would not materialize the graphs, or even if we did materialize them we might not want to ship the materialized forms.  Am I right to hope that these become unnecessary?

Reply:
Yes, long-term this should be necessary. But I suspect that it will take some time before we get to that point :)